include/mbedtls/build_info.h

```diff
@@ -112,6 +112,10 @@
 #undef MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED
 #undef MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED
 #undef MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED
+#endif
+
+#if !defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED) && \
+    !defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED)
```

Comment:
I missed that before but with this we need to change the documentation of MBEDTLS_SSL_EARLY_DATA in mbedtls_config.h.
Replace:
```
* Requires: MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED or                 
*           MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED          
*                                                                               
* Comment this to disable support for early data. If MBEDTLS_SSL_PROTO_TLS1_3   
* is not enabled, this option does not have any effect on the build.
*
```
by something like
```                                                                         
* Comment this to disable support for early data. If MBEDTLS_SSL_PROTO_TLS1_3   
* is not enabled or both MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED and
* MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED are disabled this
* option does not have any effect on the build.
```
Otherwise I think we mentioned to add MBEDTLS_SSL_SESSION_TICKETS as a requisite to MBEDTLS_SSL_EARLY_DATA but I do not see it in this PR yet.